Conceptpower+Spring/src/test/java/edu/asu/conceptpower/core/impl/ConceptTypesManagerTest.java

```diff
@@ -0,0 +1,69 @@
+package edu.asu.conceptpower.core.impl;
+
+import static org.junit.Assert.*;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.conceptpower.core.ConceptType;
+import edu.asu.conceptpower.db4o.TypeDatabaseClient;
+
+public class ConceptTypesManagerTest {
+
+    @Mock
+    private TypeDatabaseClient client = Mockito.mock(TypeDatabaseClient.class);
+
+    @InjectMocks
+    private ConceptTypesManager conceptTypesManager;
+
+    private String id = "CONCEPT_ID";
+
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+
+        ConceptType type = new ConceptType();
+        type.setTypeId("TYPE_ID");
+        type.setTypeName("TYPE_NAME");
+
+        ConceptType type2 = new ConceptType();
+        type2.setTypeId("Type Id 2");
+        type2.setTypeName("Type Name 2");
+
+        ConceptType[] types = new ConceptType[2];
+        types[0] = type;
+        types[1] = type2;
+        Mockito.when(client.getAllTypes()).thenReturn(types);
+
+        Mockito.when(client.getType(type.getTypeId())).thenReturn(type);
+        // Mockito.when(client.update(type)).thenReturn(type);
+
+        // Mockito.when(uuid.randomUUID().toString()).thenReturn("TYPE_ID");
+    }
+
+    @Test
+    public void addConceptType() {
+
+        // Not able to mock UUID class since it is a final class.

```

Comment:
You don't need to mock the uuid class. Just call the addConceptType method and then make sure that the type you added has an ID and that the addType method on the TypeDatabaseClient has been called:
https://mockito.googlecode.com/hg-history/1.5/javadoc/org/mockito/Mockito.html
(see 4. Verifying exact number of invocations / at least once / never)